cmake/tests.cmake

```diff
@@ -6,6 +6,10 @@ option(protobuf_ABSOLUTE_TEST_PLUGIN_PATH
   "Using absolute test_plugin path in tests" ON)
 mark_as_advanced(protobuf_ABSOLUTE_TEST_PLUGIN_PATH)
 
+# The compile_proto_file macro places files in the source tree, so this option
+# must be turned off.
+set(CMAKE_DISABLE_SOURCE_CHANGES OFF)
```

Comment:
Can you say more about why this change is needed here?

Reply:
If I comment out this line, I get the following error during configuration from CMake:
```
CMake Error at cmake/tests.cmake:104 (add_custom_command):
  OUTPUT path

    G:/...../src/google/protobuf/map_lite_unittest.pb.cc

  in a source directory as an output of custom command.
```

This is because the `compile_proto_file` macro adds a custom command that generates output directly into the source tree (which those files expect). My guess is that this has to do with a new check in newer versions of CMake. The failure occurs with CMake version 3.22.2.

I can add more details to the comment block, if that would help.